### PR TITLE
added book logic

### PR DESCRIPTION
**Pull request by Kate892F**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -56,7 +56,11 @@
 			<artifactId>h2</artifactId>
 			<scope>test</scope>
 		</dependency>
-
+		<dependency>
+			<groupId>org.apache.commons</groupId>
```
Comment (Elena-Bruyako) [inline pom.xml]: remove dependency

File: src/main/java/com/bookstore/model/Book.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+import java.math.BigDecimal;
+import lombok.Data;
+
+@Entity
+@Data
```
Comment (Elena-Bruyako) [inline src/main/java/com/bookstore/model/Book.java]:
```suggestion
@Getter
@Setter
```

File: src/main/java/com/bookstore/model/Book.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+@Table(name = "Books")
```
Comment (Elena-Bruyako) [inline src/main/java/com/bookstore/model/Book.java]:
```suggestion
@Table(name = "books")
```

File: src/main/java/com/bookstore/repository/impl/BookRepositoryImpl.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+import com.bookstore.repository.BookRepository;
+import com.bookstore.repository.exception.DataProcessingException;
+import java.util.List;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@Repository
```
Comment (Elena-Bruyako) [inline src/main/java/com/bookstore/repository/impl/BookRepositoryImpl.java:13]:
```suggestion
@Repository
@RequiredArgsConstructor
```

File: src/main/java/com/bookstore/repository/impl/BookRepositoryImpl.java [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+public class BookRepositoryImpl implements BookRepository {
+
+    private final SessionFactory sessionFactory;
+
+    @Autowired
+    public BookRepositoryImpl(SessionFactory sessionFactory) {
+        this.sessionFactory = sessionFactory;
+    }
```
Comment (Elena-Bruyako) [inline src/main/java/com/bookstore/repository/impl/BookRepositoryImpl.java]:
```suggestion
```

File: src/main/java/com/bookstore/service/impl/BookServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,29 @@
+package com.bookstore.service.impl;
+
+import com.bookstore.model.Book;
+import com.bookstore.repository.BookRepository;
+import com.bookstore.service.BookService;
+import java.util.List;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
```
Comment (Elena-Bruyako) [inline src/main/java/com/bookstore/service/impl/BookServiceImpl.java:10]:
```suggestion
@Service
@RequiredArgsConstructor
```

File: src/main/java/com/bookstore/service/impl/BookServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+public class BookServiceImpl implements BookService {
+
+    private final BookRepository bookRepository;
+
+    @Autowired
+    public BookServiceImpl(BookRepository bookRepository) {
+        this.bookRepository = bookRepository;
+    }
```
Comment (Elena-Bruyako) [inline src/main/java/com/bookstore/service/impl/BookServiceImpl.java]:
```suggestion
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Good job! Let's fix some comments

**Review by liliia-ponomarenko — APPROVED** [review]
LGTM!

**Pull request merged**